.github/workflows/integrate.yaml

```diff
@@ -24,8 +24,24 @@ jobs:
     - run: |
         sudo apt update
         sudo apt install tox
+    - name: Setup TFLint
+      uses: terraform-linters/setup-tflint@v4
     - run: tox -vve argo-controller-lint
 
+  terraform-check:
+    name: Check Terraform
+    uses: canonical/sdcore-github-workflows/.github/workflows/terraform.yaml@v1.0.0
+    with:
+      working-directory: charms/argo-controller/terraform
+
+  terraform-deploy:
```

Comment:
Have you considered placing the check, deploy, and lint jobs in the same re-usable workflow, e.g. `canonical/charmed-kubeflow-workflows/.github/workflows/terraform-checks.yaml`? That way everything about Terraform is abstracted there and in the case we have to change anything, we do it in a single place instead than in each charm repo.

Reply:
That's a good point actually. Let me do it.

Reply:
Do you think that we should separate also linting from the `lint` environment and move it to a separate one (`tflint`) so it's triggered only by the `terraform-checks.yaml` file? Because right now, it runs as part of `lint` and `lib-check` jobs.

Reply:
Done what I 've described above

Reply:
https://github.com/canonical/charmed-kubeflow-workflows/pull/61